src/main/java/uk/gov/hmcts/cmc/claimstore/services/SettlementAgreementService.java

```diff
@@ -65,18 +74,32 @@ public Claim countersign(Claim claim, String authorisation) {
         Settlement settlement = assertSettlementCanBeResponded(claim);
         settlement.countersign(MadeBy.DEFENDANT);
 
-        String userAction = format("SETTLEMENT_AGREEMENT_COUNTERSIGNED_BY_%s", MadeBy.DEFENDANT.name());
+        String userAction = AGREEMENT_COUNTER_SIGNED_BY_DEFENDANT.getValue();
         caseRepository.updateSettlement(claim, settlement, authorisation, userAction);
 
         Claim updated = claimService.getClaimByExternalId(claim.getExternalId(), authorisation);
 
         eventProducer.createSettlementAgreementCountersignedEvent(updated);
-
+        this.ccdEventProducer.createCCDSettlementEvent(claim, settlement, authorisation, userAction);
         AppInsightsEvent appInsightsEvent = settlement.isSettlementThroughAdmissions()
             ? SETTLEMENT_AGREEMENT_REACHED_BY_ADMISSION : SETTLEMENT_AGREEMENT_REACHED;
         appInsights.trackEvent(appInsightsEvent, REFERENCE_NUMBER, updated.getReferenceNumber());
 
         return updated;
+
+    }
+
+    protected void signSettlementAgreement(String externalId, Settlement settlement, String authorisation) {
+        final Claim claim = claimService.getClaimByExternalId(externalId, authorisation);
```

Comment:
minor: do we need the `final` and `this.` here?

Reply:
please remove all final from method, we cleaned final usage from every where.

Reply:
@dharmendrak , @markry-kainos and @sabahirfan this was refactor from offerservice into this service, just lift and shift. (not my code) but Will sort it out.